### ISSUE TITLE
Refactor origin project creation API

### ISSUE THREAD

**Issue by chefsalim** [habitat-sh/habitat#4141]
We recently made some changes to the origin project creation API to make it depend on passing in an installation id and repo id into the json, and removing the ability to pass in a repo by name.  There are two issues with this approach: 1) we add a hard dependency on the specific github auth (in this case, the github app installation id) into the API schema, and 2) we broke the bulk project creation tool (and the builder dev project creation flow). 

For reference, the original JSON used to be as per the below, and a project could be created via a POST with the personal token. 
```
{
    "origin": "core",
    "plan_path": "nginx/plan.sh",
    "github": {
        "organization": "habitat-sh",
        "repo": "core-plans"
    }
}


http POST http://localhost:9636/v1/projects Authorization:Bearer:${HAB_AUTH_TOKEN} < project.json
```

**Comment by chefsalim** [comment]
We recently made some changes to the origin project creation API to make it depend on passing in an installation id and repo id into the json, and removing the ability to pass in a repo by name.  There are two issues with this approach: 1) we add a hard dependency on the specific github auth (in this case, the github app installation id) into the API schema, and 2) we broke the bulk project creation tool (and the builder dev project creation flow). 

For reference, the original JSON used to be as per the below, and a project could be created via a POST with the personal token. 
```
{
    "origin": "core",
    "plan_path": "nginx/plan.sh",
    "github": {
        "organization": "habitat-sh",
        "repo": "core-plans"
    }
}


http POST http://localhost:9636/v1/projects Authorization:Bearer:${HAB_AUTH_TOKEN} < project.json
```
